eclipse-collections-code-generator/src/main/resources/impl/list/mutable/synchronizedPrimitiveList.stg

```diff
@@ -217,6 +217,15 @@ public class Synchronized<name>List
         }
     }
 
+    @Override
+    public MutableList\<<wrapperName>\> boxed()
+    {
+        synchronized (this.getLock())
+        {
+            return new BoxedMutable<name>List(this).asSynchronized();
+        }
+    }
+
```

Comment:
I think we need to share a lock here, because it's a view on a view. A similar situation happens in SynchronizedMutableMap, when we need a keySet().

```java
    @Override
    public Set<K> keySet()
    {
        synchronized (this.lock)
        {
            return SynchronizedMutableSet.of(this.getDelegate().keySet(), this.lock);
        }
    }
```

Reply:
Modified. Thanks.